frontend/src/pages/modelServing/screens/projects/ModelServingPlatform.tsx

```diff
@@ -0,0 +1,145 @@
+    servingRuntimeTemplateOrder: { data: templateOrder },
+    servingRuntimeTemplateDisablement: { data: templateDisablement },
+    serverSecrets: { refresh: refreshTokens },
+    inferenceServices: { refresh: refreshInferenceServices },
+    currentProject,
+  } = React.useContext(ProjectDetailsContext);
+
+  const { refresh: refreshAllProjects } = React.useContext(ProjectsContext);
+
+  const templatesSorted = getSortedTemplates(templates, templateOrder);
+  const templatesEnabled = templatesSorted.filter((template) =>
+    getTemplateEnabled(template, templateDisablement),
+  );
+
+  const emptyTemplates = templatesEnabled.length === 0;
+  const emptyModelServer = servingRuntimes.length === 0;
+
+  const currentProjectServingPlatform = getProjectModelServingPlatform(
+    currentProject,
+    disableKServe,
+    disableModelMesh,
+  );
+
+  const isProjectModelMesh = currentProjectServingPlatform === ServingRuntimePlatform.MULTI;
+
+  const shouldShowPlatformSelection =
+    !disableKServe && !disableModelMesh && !currentProjectServingPlatform;
+
+  return (
+    <>
+      <DetailsSection
+        id={ProjectSectionID.MODEL_SERVER}
+        title={ProjectSectionTitles[ProjectSectionID.MODEL_SERVER]}
+        actions={
+          shouldShowPlatformSelection
+            ? undefined
+            : [
+                <ModelServingPlatformButtonAction
+                  isProjectModelMesh={isProjectModelMesh}
+                  emptyTemplates={emptyTemplates}
+                  onClick={() => {
+                    if (isProjectModelMesh) {
+                      setOpen(true);
+                    }
+                    // else, show the kserve deploy model modal
+                  }}
+                  key="serving-runtime-actions"
+                />,
+              ]
+        }
+        isLoading={!servingRuntimesLoaded && !templatesLoaded}
+        isEmpty={!shouldShowPlatformSelection && emptyModelServer}
+        loadError={servingRuntimeError || templateError}
+        emptyState={
+          <EmptyDetailsList
+            title={isProjectModelMesh ? 'No model servers' : 'No deployed models'}
+            description={
+              isProjectModelMesh
+                ? 'Before deploying a model, you must first add a model server.'
+                : 'To get started, deploy a model.'
+            }
+            icon={PlusCircleIcon}
+          />
+        }
+        labels={
+          currentProjectServingPlatform && [
+            <Label key="serving-platform-label">
+              {isProjectModelMesh ? 'Multi-model serving enabled' : 'Single model serving enabled'}
+            </Label>,
+          ]
+        }
+      >
+        {shouldShowPlatformSelection ? (
+          <ModelServingPlatformSelect
+            onSelect={(selectedPlatform) => {
+              if (selectedPlatform === ServingRuntimePlatform.MULTI) {
+                setOpen(true);
+              }
```

Comment:
You can prepare for the next update by supporting the selected platform being MULTI or SINGLE and setting it as the open state.

Reply:
I will leave this for @lucferbux since he is working on the KServe modal.